qt/scientific_interfaces/test/EnggDiffGSASFittingPresenterTest.h

```diff
@@ -0,0 +1,349 @@
+#ifndef MANTID_CUSTOM_INTERFACES_ENGGDIFFGSASFITTINGPRESENTERTEST_H_
+#define MANTID_CUSTOM_INTERFACES_ENGGDIFFGSASFITTINGPRESENTERTEST_H_
+
+#include "../EnggDiffraction/EnggDiffGSASFittingPresenter.h"
+#include "EnggDiffGSASFittingModelMock.h"
+#include "EnggDiffGSASFittingViewMock.h"
+
+#include "MantidAPI/WorkspaceFactory.h"
+#include "MantidKernel/make_unique.h"
+#include "MantidTestHelpers/WorkspaceCreationHelper.h"
+
+#include <cxxtest/TestSuite.h>
+
+using namespace MantidQt::CustomInterfaces;
+using testing::Return;
+using testing::Throw;
+
+class EnggDiffGSASFittingPresenterTest : public CxxTest::TestSuite {
+
+public:
+  void test_loadValidFile() {
+    auto mockModel = Mantid::Kernel::make_unique<
+        testing::NiceMock<MockEnggDiffGSASFittingModel>>();
+    auto mockModel_ptr = mockModel.get();
+
+    auto mockView = Mantid::Kernel::make_unique<
+        testing::NiceMock<MockEnggDiffGSASFittingView>>();
+    auto mockView_ptr = mockView.get();
+
+    EnggDiffGSASFittingPresenter pres(std::move(mockModel),
+                                      std::move(mockView));
```

Comment:
The lines 22-31 (setup) and 53-58 (assertions) seem to be repeated for all the tests. I wonder if it is a better idea to separate them out in setUp and doTest?